tests/unit/test_eurostat.py

```diff
@@ -0,0 +1,223 @@
+"""'test_eurostat.py'."""
+
+import json
+import logging
+
+import pandas as pd
+import pytest
+from viadot.sources import Eurostat
+
+
+class EurostatMock(Eurostat):
+    """Mock of Eurostat source class."""
+
+    def __init__(self, dataset_code=None):
+        # Inicjalizacja bez dataset_code
+        super().__init__(dataset_code=dataset_code)
+
+    def _download_json(self):
+        with open("test_eurostat_response.json") as file:
+            data = json.load(file)
+            return data
+
+
+def test_eurostat_dictionary_to_df():
+    """Test eurostat_dictionary_to_df method from source class."""
+    eurostat = EurostatMock(dataset_code="")  # Możesz przekazać pusty string lub None
+    data = eurostat._download_json()
+
+    result_df = eurostat.eurostat_dictionary_to_df(["geo", "time"], data)
+
+    assert list(result_df.columns) == ["geo", "time", "indicator"]
+
+    expected_years = ["2020", "2021", "2022"]
+    assert result_df["time"].unique().tolist() == expected_years
+
+    expected_geo = ["Germany", "France", "Italy"]
+    assert result_df["geo"].unique().tolist() == expected_geo
+
+    expected_indicator = [
+        100,
+        150,
+        200,
+        110,
+        160,
+        210,
+        120,
+        170,
+        220,
+    ]
+    assert result_df["indicator"].unique().tolist() == expected_indicator
+
+URL = (
+    "https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0"
+    "/data/ILC_DI04?format=JSON&lang=EN"
+)
```

Comment:
This should be at the top of the module

Reply:
[7824be9](https://github.com/dyvenia/viadot/pull/1041/commits/7824be98f76a5daf19f8dadf9eba3c8147bbdee8)